core/src/test/java/org/infinispan/executors/LimitedExecutorTest.java

```diff
@@ -57,11 +57,13 @@ public void testConcurrencyLimit() throws Exception {
       limitedExecutor.execute(() -> cf2.complete("bla"));
 
       Thread.sleep(10);
+      assertFalse(cf1.isDone());
       assertFalse(cf2.isDone());
 
       latch.countDown();
       assertEquals("bla", cf1.get(10, SECONDS));
       assertEquals("bla", cf2.get(10, SECONDS));
+      eventuallyEquals(0, executor::getActiveCount);
```

Comment:
I've added javadocs to the test methods. `testConcurrencyLimit()` uses regular `execute()`, so it needs 2 threads to check that one task really blocks the other. This test method doesn't need 2 threads, because it uses `executeAsync()`, and the 1st task blocks the executor without using any threads.